testing/hooks.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+type THookFn = () => void | Promise<void>;
+type THook = (fn: THookFn) => void | Promise<void>;
+type HookType = "beforeEach" | "afterEach";
+
+const hooks: Partial<Record<HookType, THookFn>> = {};
```

Comment:
This is needlessly complicated.
Just define an object with well known members to contain the state.
Also should probably be arrays.

```suggestion
const hooks = {
  beforeEach: [],
  afterEach: [],
};
```

Reply:
Array means that user can add more beforeEach and afterEach